src/main/resources/interface-specification.yml

```diff
@@ -7796,6 +7886,62 @@ components:
           $ref: '#/components/schemas/AgreementApprovalPolicy'
         eservice:
           $ref: '#/components/schemas/CatalogDescriptorEService'
+    ClientDetails:
+      description: Models Client details
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+        consumer:
+          $ref: '#/components/schemas/Organization'
+        name:
+          type: string
+        purposes:
+          type: array
+          items:
+            $ref: '#/components/schemas/ClientPurpose'
+        description:
+          type: string
+        kind:
+          $ref: '#/components/schemas/ClientKind'
+      required:
+        - id
+        - organization
+        - consumer
+        - name
+        - purposes
+        - kind     
+    ClientKind:
+      type: string
+      description: kind of client
+      enum:
+        - CONSUMER
+        - API
+    ClientPurpose:
+      type: object
+      properties:
+        purposeId:
+          type: string
+          format: uuid
+        title:
+          type: string
+        eservice:
+          $ref: '#/components/schemas/CompactEServiceLight'  
```

Comment:
let's confirm it with the frontend, but we should maybe use `CompactEService` here (which has the producer field)